io/seqio/fasta/example_fasta_test.go

```diff
@@ -0,0 +1,43 @@
+package fasta_test
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/biogo/biogo/alphabet"
+	"github.com/biogo/biogo/io/seqio"
+	"github.com/biogo/biogo/io/seqio/fasta"
+	"github.com/biogo/biogo/seq/linear"
+)
+
+func ExampleReader() {
+	const multiFasta = `>SequenceA
+GATC
+>SequenceB
+CTAG
+>SequenceC
+CTaaaG`
+
+	data := strings.NewReader(multiFasta)
+	template := linear.NewSeq("", nil, alphabet.DNAredundant)
+	r := fasta.NewReader(data, template)
+	scanner := seqio.NewScanner(r)
+
+	// Iterate through each sequence in a multifasta
+	for scanner.Next() {
+		sequence := scanner.Seq()
+		asString := sequence.(*linear.Seq).Seq.String()
+		asBytes := alphabet.LettersToBytes(sequence.(*linear.Seq).Seq)
+		fmt.Printf("%q as %T: %s and %T: %v\n", sequence.Name(), asString, asString, asBytes, asBytes)
+		// ./example_fasta_test.go:29:3: Printf format %a has unknown verb a
+		//fmt.Printf("%a\n", sequence)
+	}
+	if err := scanner.Error(); err != nil {
+		fmt.Println(fmt.Errorf("failed during first read: %w", err))
+	}
+
+	// Output:
+	// "SequenceA" as string: GATC and []uint8: [71 65 84 67]
+	// "SequenceB" as string: CTAG and []uint8: [67 84 65 71]
+	// "SequenceC" as string: CTaaaG and []uint8: [67 84 97 97 97 71]
```

Comment:
```
	data := strings.NewReader(multiFasta)
	template := linear.NewSeq("", nil, alphabet.DNAredundant)
	r := fasta.NewReader(data, template)
	sc := seqio.NewScanner(r)

	// Iterate through each sequence in a multi-fasta, printing the sequence.
	for sc.Next() {
		s := sc.Seq()
		fmt.Println(s)
	}
	if err := sc.Error(); err != nil {
		fmt.Printf("failed during read: %v", err)
	}

	// Output:
	// "SequenceA" GATC
	// "SequenceB" CTAG
	// "SequenceC" CTaaaG
```

Reply:
This looks the same as it was.

Reply:
A problem this example aims to address is showing how the sequence name and value can be accessed separately and as basic concrete types. For a friend of mine who was trying to use biogo, there was too much abstraction.

Reply:
OK. That should be an example on the sequence types then. This is an example for the fasta reader.

Reply:
It is non-obvious how to get concrete types from the fasta reader. How would you get the nucleotide sequence apart from the sequence name?

Reply:
Ah, OK. I will have a think about this. ISTM that this is a Go language learning issue rather than a bíogo learning issue.

Reply:
```
	const multiFasta = `
	>SequenceA dam methylation site
	GATC
	>SequenceB ori motif
	CTAG
	>SequenceC CTCF binding motif
	CCGCGNGGNGGCAG`

	data := strings.NewReader(multiFasta)

	// fasta.Reader requires a known type template to fill
	// with FASTA data. Here we use *linear.Seq.
	template := linear.NewSeq("", nil, alphabet.DNAredundant)
	r := fasta.NewReader(data, template)

	// Make a seqio.Scanner to simplify iterating over a
	// stream of data.
	sc := seqio.NewScanner(r)

	// Iterate through each sequence in a multifasta and examine the
	// annotation data.
	for sc.Next() {
		// Get the current sequence and type assert to *linear.Seq.
		// While this is unnecessary here, it can be useful to have
		// the concrete type.
		s := sc.Seq().(*linear.Seq)

		// Print the sequence ID, description and sequence data.
		fmt.Printf("%q %q %s\n", s.ID, s.Desc, s.Seq)
	}
	if err := sc.Error(); err != nil {
		log.Fatal(err)
	}

	// Output:
	// "SequenceA" "dam methylation site" GATC
	// "SequenceB" "ori motif" CTAG
	// "SequenceC" "CTCF binding motif" CCGCGNGGNGGCAG
```